src/test/java/com/github/paweladamski/httpclientmock/HttpClientVerifyTest.java

```diff
@@ -1,288 +1,288 @@
+
+    httpClientMock.execute(new HttpGet("http://example.com"));
+    httpClientMock.execute(new HttpGet("http://example.com"));
+    httpClientMock.execute(new HttpGet("http://example.com"));
+
+    httpClientMock.verify()
+        .get("http://localhost")
+        .called();
+    httpClientMock.verify()
+        .get("http://www.google.com")
+        .called(2);
+    httpClientMock.verify()
+        .get("http://example.com")
+        .called(3);
+  }
+
+  @Test
+  public void shouldVerifyBodyContent() throws IOException {
+    HttpClientMock httpClientMock = new HttpClientMock();
+
+    httpClientMock.execute(httpPost("http://localhost", "foo"));
+    httpClientMock.execute(httpPost("http://localhost", "foo"));
+
+    httpClientMock.execute(httpPut("http://localhost", "bar"));
+    httpClientMock.execute(httpPut("http://localhost", "foo"));
+
+    httpClientMock.verify()
+        .post("http://localhost")
+        .withBody(containsString("foo"))
+        .called(2);
+    httpClientMock.verify()
+        .put("http://localhost")
+        .withBody(containsString("bar"))
+        .called();
+    httpClientMock.verify()
+        .get("http://localhost")
+        .withBody(containsString("foo bar"))
+        .notCalled();
+  }
+
+  @Test
+  public void should_handle_path_with_query_parameter() throws IOException {
+    HttpClientMock httpClientMock = new HttpClientMock();
+
+    httpClientMock.execute(httpPost("http://localhost?a=1&b=2&c=3"));
+    httpClientMock.execute(httpPost("http://localhost?a=1&b=2"));
+    httpClientMock.execute(httpPost("http://localhost?a=1"));
+
+    httpClientMock.verify()
+        .post("http://localhost?d=3")
+        .notCalled();
+    httpClientMock.verify()
+        .post("http://localhost?a=3")
+        .notCalled();
+    httpClientMock.verify()
+        .post("http://localhost?a=1&b=2&c=3")
+        .called(1);
+    httpClientMock.verify()
+        .post("http://localhost?a=1&b=2")
+        .called(1);
+    httpClientMock.verify()
+        .post("http://localhost?a=1")
+        .called(1);
+    httpClientMock.verify()
+        .post("http://localhost")
+        .withParameter("a", "1")
+        .called(1);
+
+    httpClientMock.verify()
+        .post("http://localhost")
+        .notCalled();
+  }
+
+  @Test
+  public void should_handle_path_with_reference() throws IOException {
+    HttpClientMock httpClientMock = new HttpClientMock();
+
+    httpClientMock.execute(httpPost("http://localhost?a=1#abc"));
+    httpClientMock.execute(httpPost("http://localhost#xyz"));
+
+    httpClientMock.verify()
+        .post("http://localhost?a=1#abc")
+        .called(1);
+    httpClientMock.verify()
+        .post("http://localhost#abc")
+        .notCalled();
+    httpClientMock.verify()
+        .post("http://localhost#xyz")
+        .called(1);
+    httpClientMock.verify()
+        .post("http://localhost")
+        .notCalled();
+  }
+
+  @Test(expected = AssertionError.class)

```

Comment:
As I explained in previous comment AssertionError should not be used here.

Reply:
Fixed in 7e629fe9aeb8dbbff09615e3b4e0e12c6096fa2c